filecoin-proofs/src/api/post_adapter.rs

```diff
@@ -0,0 +1,626 @@
+use crate::api::internal::ChallengeSeed;
+use crate::api::internal::Commitment;
+use crate::api::internal::POST_SECTORS_COUNT;
+use crate::error;
+use sector_base::api::bytes_amount::PaddedBytesAmount;
+use std::iter::repeat;
+use std::iter::repeat_with;
+
+pub struct GeneratePoStDynamicSectorsCountInput {
+    pub sector_bytes: PaddedBytesAmount,
+    pub challenge_seed: ChallengeSeed,
+    pub input_parts: Vec<(Option<String>, Commitment)>,
+}
+
+pub struct GeneratePoStFixedSectorsCountInput {
+    pub sector_bytes: PaddedBytesAmount,
+    pub challenge_seed: ChallengeSeed,
+    pub input_parts: [(Option<String>, Commitment); POST_SECTORS_COUNT],
+}
+
+pub struct VerifyPoStDynamicSectorsCountInput {
+    pub sector_bytes: PaddedBytesAmount,
+    pub comm_rs: Vec<Commitment>,
+    pub challenge_seed: ChallengeSeed,
+    pub proofs: Vec<[u8; 192]>,
+    pub faults: Vec<u64>,
+}
+
+pub struct VerifyPoStFixedSectorsCountInput {
+    pub sector_bytes: PaddedBytesAmount,
+    pub comm_rs: [Commitment; POST_SECTORS_COUNT],
+    pub challenge_seed: ChallengeSeed,
+    pub proof: [u8; 192],
+    pub faults: Vec<u64>,
+}
+
+pub struct GeneratePoStDynamicSectorsCountOutput {
+    pub proofs: Vec<[u8; 192]>,
+    pub faults: Vec<u64>,
+}
+
+pub struct GeneratePoStFixedSectorsCountOutput {
+    pub proof: [u8; 192],
+    pub faults: Vec<u64>,
+}
+
+pub struct VerifyPoStDynamicSectorsCountOutput {
+    pub is_valid: bool,
+}
+
+pub struct VerifyPoStFixedSectorsCountOutput {
+    pub is_valid: bool,
+}
+
+/// Maps inputs for a single dynamic sectors-count PoSt generation operation to
+/// a vector of PoSt generation inputs for fixed sectors-count.
+///
+pub fn generate_post_spread_input(
+    dynamic: GeneratePoStDynamicSectorsCountInput,
+) -> Vec<GeneratePoStFixedSectorsCountInput> {
+    let mut fixed: Vec<GeneratePoStFixedSectorsCountInput> = vec![];
+
+    let chunks = dynamic.input_parts.chunks_exact(POST_SECTORS_COUNT);
+
+    let remainder = chunks.remainder();
+
+    for chunk in chunks {
+        fixed.push(GeneratePoStFixedSectorsCountInput {
+            sector_bytes: dynamic.sector_bytes,
+            challenge_seed: dynamic.challenge_seed,
+            input_parts: [chunk[0].clone(), chunk[1].clone()],
+        });
+    }
+
+    if !remainder.is_empty() {
+        let mut input_parts: [(Option<String>, Commitment); POST_SECTORS_COUNT] =
+            Default::default();
+
+        let vs = repeat_with(|| remainder[remainder.len() - 1].clone());
+        let rs = remainder.iter().cloned();
+        let it = rs.chain(vs).take(POST_SECTORS_COUNT).enumerate();
```

Comment:
I get what's going on here, but `vs`/`rs`/`it` don't carry much meaning within my context (except `it` which I suppose it's "iterator").